### PR TITLE
feat: add support for Terraform v0.15

### PR DESCRIPTION
**Pull request by soerenmartius**
- feat: add support for Terraform `v0.15`
- feat: upgrade pre-commit hooks to `v0.2.3`
- build: upgrade build-tools hooks to `v0.11.0`
- chore: prepare v0.5.0 release
- fix: remove ssh forwarding in Makefile

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -353,7 +353,7 @@ Copyright &copy; 2021 [Mineiros GmbH][homepage]
 [badge-build]: https://github.com/mineiros-io/terraform-aws-iam-role/workflows/CI/CD%20Pipeline/badge.svg
 [badge-semver]: https://img.shields.io/github/v/tag/mineiros-io/terraform-aws-iam-role.svg?label=latest&sort=semver
 [badge-license]: https://img.shields.io/badge/license-Apache%202.0-brightgreen.svg
-[badge-terraform]: https://img.shields.io/badge/Terraform-0.13%20and%200.12.20+-623CE4.svg?logo=terraform
+[badge-terraform]: https://img.shields.io/badge/Terraform-0.15%20and%200.12.20+-623CE4.svg?logo=terraform
```
Comment (mariux) [inline README.md:356]: 13 and 14 should be added to the badge

**Review by mariux — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (soerenmartius) [reply README.md:356]: nope, doesn't look appropriate to me - there  is absolutely no value in listing every single version. We should use a semver range instead. Also, a badge shouldn't      block the PR.

Comment (mariux) [reply README.md:356]: but the badge is wrong, "0.15 and 0.12.20" is just not true..

Comment (mariux) [inline README.md:356]:
```suggestion
[badge-terraform]: https://img.shields.io/badge/terraform-0.15%20|0.14%20|%200.13%20|%200.12.20+-623CE4.svg?logo=terraform
```

Comment (soerenmartius) [reply README.md]: resolved in 439db77

Comment (soerenmartius) [reply README.md]: resolved

**Review by mariux — APPROVED** [review]
lgtm

**Pull request merged**